### PR TITLE
Introduce EDispMap class 

### PR DESCRIPTION
**Pull request by registerrier**
This PR introduces the `EDispMap`class to be used to contain the energy dispersion at each position in a `Map`. The 1D `EnergyDispersion` used to convert from the `e_true` space to `e_reco` space is obtained with the `EDispMap.get_energy_dispersion(position, e_reco)` method.
At the moment, the `stacking` method works only for maps with identical `MapGeom`.

There is significant code duplication with `PSFMap`. This might be partly solved by using inheritance from e.g. a more general `IRFMap`

### CODE REVIEW

File: gammapy/cube/edisp_map.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+def make_edisp_map(edisp, pointing, geom, max_offset, exposure_map=None):
+    """Make a edisp map for a single observation
+
+    Expected axes : migra and true energy in this specific order
+    The name of the migra MapAxis is expected to be 'migra'
+
+    Parameters
+    ----------
+    psf : `~gammapy.irf.EnergyDispersion2D`
```
Comment (adonath) [inline gammapy/cube/edisp_map.py]: psf -> edisp

File: gammapy/cube/edisp_map.py [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+
+class EDispMap(object):
+    """Class containing the Map of Energy Dispersions and allowing to interact with it.
+
+    Parameters
+    ----------
+    edisp_map : `~gammapy.maps.Map`
+        the input Energy Dispersion Map. Should be a Map with 2 non spatial axes.
+        migra and true energy axes should be given in this specific order.
+
```
Comment (adonath) [inline gammapy/cube/edisp_map.py]: Add `exposure_map` to the docstring...

File: gammapy/cube/edisp_map.py [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+        hdulist.writeto(filename, overwrite=overwrite)
+
+    def get_energy_dispersion(self, position, e_reco, migra_step=5e-3):
+        """ Returns EnergyDispersion at a given position
+
+        Parameters
+        ----------
+        position : `~astropy.coordinates.SkyCoord`
+            the target position. Should be a single coordinates
+
```
Comment (adonath) [inline gammapy/cube/edisp_map.py]: Add `e_reco` and  `migra_step` here...

File: gammapy/cube/edisp_map.py [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+        Returns
+        -------
+        psf_table : `~gammapy.irf.EnergyDependentTablePSF`
```
Comment (adonath) [inline gammapy/cube/edisp_map.py]: The return type is not correctly documented, change to `EnergyDispersion`

File: gammapy/cube/tests/test_edisp_map.py [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        name="energy",
+        node_type="edges",
+        interp="log",
+    )
+    migra_axis = MapAxis(nodes=np.linspace(0.0, 3.0, 51), unit="", name="migra")
+
+    edisp2d = EnergyDispersion2D.from_gauss(etrue, migra, 0.0, 0.2, offsets)
+
+    geom = WcsGeom.create(
+        skydir=pointing, binsz=0.2, width=5, axes=[migra_axis, energy_axis]
```
Comment (adonath) [inline gammapy/cube/tests/test_edisp_map.py]: Could you use a smaller map for testing? Let's say 5x5 spatial pixels instead of 25 x 25?

**Comment by adonath** [comment]
@registerrier I have some ideas how to simplify `.get_energy_dispersion()`. If you're OK with that I could do a follow-up PR next week, once this one is merged. So we should just make sure that the test coverage is good enough...

File: gammapy/cube/tests/test_edisp_map.py (hunk 5 above)
Comment (registerrier) [reply gammapy/cube/tests/test_edisp_map.py]: Done

File: gammapy/cube/edisp_map.py (hunk 3 above)
Comment (registerrier) [reply gammapy/cube/edisp_map.py]: OK corrected

File: gammapy/cube/edisp_map.py (hunk 1 above)
Comment (registerrier) [reply gammapy/cube/edisp_map.py]: done

**Pull request merged**